tests/moduletool/test_install.py

```diff
@@ -893,6 +911,127 @@ def test_install_from_index_dont_leak_pip_index(
     assert os.getenv(env_var) == index.url if env_var != "PIP_CONFIG_FILE" else pip_config_file
 
 
+@pytest.mark.parametrize_any("use_pip_config", [True, False])
+def test_install_with_use_config(
+    tmpdir: py.path.local,
+    modules_v2_dir: str,
+    snippetcompiler_clean,
+    monkeypatch,
+    use_pip_config,
+    caplog,
+) -> None:
+    """
+    Test that when using "use_pip_config" the configfile is used and that the module is being installed and
+    that no package source needs to be set.
+    if "use_pip_config" is not used, verify that it will install the module from the specified package source
+    """
+    index: PipIndex = PipIndex(artifact_dir=os.path.join(str(tmpdir), ".custom-index"))
+    # prepare v2 modules
+    v2_template_path: str = os.path.join(modules_v2_dir, "minimalv2module")
+    v2mod1: module.ModuleV2Metadata = module_from_template(
+        v2_template_path,
+        os.path.join(str(tmpdir), "v2mod1"),
+        new_name="v2mod1",
+        publish_index=index,
+    )
+
+    pip_config_file = os.path.join(tmpdir, "pip.conf")
+    with open(pip_config_file, "w+", encoding="utf-8") as f:
+        f.write("[global]\n")
+        f.write("timeout = 60\n")
+        f.write("index-url = file://" + index.url + "\n")
+    monkeypatch.setenv("PIP_CONFIG_FILE", pip_config_file)
```

Comment:
I'm still missing a test case that doesn't have `PIP_CONFIG_FILE` set (monkeypatch delete it from the environment). This test ideally shouldn't try to install a non-existing module though because that could open us op to dependency confusion attacks.

Reply:
You mean if PIP_CONFIG_FILE is not set it will use the default and will try to get the module from pypi and so I should use an existing module like inmanta-module-dummy-module?

Reply:
I tried it like this:
```
def test_install_with_use_config_but_PIP_CONFIG_FILE_not_set(    tmpvenv_active_inherit: env.VirtualEnv,
    modules_v2_dir: str,
    snippetcompiler_clean,
    monkeypatch,
    caplog,
) -> None:
    monkeypatch.delenv("PIP_CONFIG_FILE", False)

    # set up project
    snippetcompiler_clean.setup_for_snippet(
        """
        import dummy_module
        """,
        autostd=False,
        install_project=False,
        use_pip_config_file=True,
    )

    # install project
    project_path = module.Project.get().path
    os.chdir(project_path)
    with caplog.at_level(logging.DEBUG):
        ProjectTool().execute("install", [])
    print(caplog.text)
    assert tmpvenv_active_inherit.are_installed(requirements=["inmanta-module-dummy-module"])
```

but the dummy gets installed as a V1 module. I Don't know how to fix this. Is it something wrong with my setup_for_snippet or should I change `repo: https://github.com/inmanta/` in `project.yml` to something else?

Reply:
You forgot to add it to the `python_requires`. For the other tests you did do so.

Reply:
indeed, sorry for that I don't know how I missed it

Reply:
No problem, it happens!